nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ConvertJSONtoCSV.java

```diff
@@ -0,0 +1,360 @@
+        "and the combination of the two regardless of the level of nesting in the JSON document.")
+public class ConvertJSONtoCSV extends AbstractProcessor {
+    volatile String delimiter;
+    volatile String removeFields;
+    volatile String emptyFields = "";
+
+    public static final AllowableValue INCLUDE_HEADER_TRUE = new AllowableValue(
+            "True", "True", "Creates headers for each JSON file.");
+    public static final AllowableValue INCLUDE_HEADER_FALSE = new AllowableValue(
+            "False", "False", "Only parses the JSON fields and does not include headers");
+
+
+    public static final PropertyDescriptor DELIMITER = new PropertyDescriptor
+            .Builder().name("CSV Delimiter")
+            .description("Delimiter used for the generated CSV output (Example: , | -)")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor REMOVE_FIELDS = new PropertyDescriptor
+            .Builder().name("Remove JSON Fields/Columns")
+            .description("Comma delimited list of columns that should be removed when parsing JSON and building the CSV. " +
+                    "This includes all top level and most granular nested fields/columns. By default with nothing specified every field " +
+                    "will be parsed.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor EMPTY_FIELDS = new PropertyDescriptor
+            .Builder().name("Empty field value")
+            .description("During denormalization/flattening of the JSON the value that will be substituted for empty fields values " +
+                    "(Example: NULL). Defaults to empty string if not specified.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor INCLUDE_HEADERS = new PropertyDescriptor
+            .Builder().name("Include Headers")
+            .description("Whether or not to include headers in the CSV output")
+            .required(true)
+            .allowableValues(INCLUDE_HEADER_TRUE, INCLUDE_HEADER_FALSE)
+            .defaultValue(INCLUDE_HEADER_TRUE.getValue())
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final Relationship RELATIONSHIP_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Successfully parsing the JSON file to CSV ")
+            .build();
+
+    public static final Relationship RELATIONSHIP_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("Failed parsing the JSON file to CSV ")
+            .build();
+
```

Comment:
Should there be an "original" relationship that the JSON file would be transferred to upon success? Other processors do similar things.